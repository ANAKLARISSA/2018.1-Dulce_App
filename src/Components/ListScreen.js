import React from 'react';
import {Container, Content, List, ListItem, Text} from 'native-base';
import {View} from 'react-native';
import ScreenHeader from '../Components/ScreenHeader';

class ListScreen extends React.Component {

  render() {
    return (
      <Container>
      <View>
      <ScreenHeader
        title={this.props.title}
      />
      </View>
        <Content>
          <List dataArray={this.props.list}
            renderRow={(item) => (
              <ListItem onPress={() => this.props.onPress()}>
                <Text>{item}</Text>
              </ListItem>
            )}
          />
        </Content>
      </Container>
    );
  }
}

export default ListScreen;
