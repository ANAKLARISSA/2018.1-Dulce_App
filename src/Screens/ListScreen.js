import React from 'react';
import {ScrollView, Alert, FlatList} from 'react-native';
import axios from 'axios';
import store from '../Reducers/store';
import {Container, Content, Spinner, ListItem, Left, Body, Text, Title, Header} from 'native-base';
import UserAvatar from 'react-native-user-avatar';
import ScreenHeader from '../Components/ScreenHeader';

class ListScreen extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      employees: [],
      loading: true
    };
  }

  componentDidMount() {
    const url = 'http://172.18.0.1:8083/api/userManager/listUser';
    this.setState({loading: true});
    axios.get(url, {
      headers: {
        'x-access-token': store.getState().currentUser.token
      }
    })
      .then((response) => {
        this.setState({employees: response.data, loading: false});
      })
      .catch(() => {Alert.alert('ERRO');});
  }

  render() {
    return (
      <ScrollView>
      <ScreenHeader title='Lista de Médicos'/>
        {this.state.loading ? (
            <Container>
              <Content>
                <Spinner color= '#5f4b8b'/>
              </Content>
            </Container>
          ) : (
            <FlatList
              data = {this.state.employees}
              keyExtractor = {(item) => {return item.id.toString();}}
              renderItem={(data) => {return (
                <ListItem onPress={() => this.props.navigation.navigate('profile', {userId: data.item.id})} key={data.item.id} avatar>
                  <Left> <UserAvatar size='50' name={data.item.name} /></Left>
                  <Body><Text>{data.item.name}</Text>
                    <Text note>Matrícula: {data.item.registration}</Text></Body>
                </ListItem>
              );}}/>
          )}
      </ScrollView>
    );
  }
}
export default ListScreen;
