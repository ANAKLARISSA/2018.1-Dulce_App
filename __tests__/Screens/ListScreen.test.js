import 'react-native';
import React from 'react';
import ListScreen from '../../src/Screens/ListScreen';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
require('bezier');
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
  .create(<ListScreen />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

it('UserList snapshot test', async () => {
  const wrapper = shallow(<ListScreen />);
  await flushPromises();
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});
