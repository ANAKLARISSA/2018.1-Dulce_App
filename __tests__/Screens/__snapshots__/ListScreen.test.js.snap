// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ScreenHeader
          title="Lista de Médicos"
        />,
        <FlatList
          data={
            Object {
              "message": "404 data not found",
            }
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "Lista de Médicos",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "message": "404 data not found",
          },
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "children": Array [
              <ScreenHeader
                title="Lista de Médicos"
              />,
              <Styled(Container)
                virtual={undefined}
              >
                <Styled(Content)
                  virtual={undefined}
                >
                  <Styled(Spinner)
                    color="#5f4b8b"
                    virtual={undefined}
                  />
                </Styled(Content)>
              </Styled(Container)>,
            ],
          },
          Object {},
        ],
        Array [
          Object {
            "automaticallyAdjustContentInsets": false,
            "children": <Styled(Spinner)
              color="#5f4b8b"
              virtual={undefined}
            />,
            "contentContainerStyle": Array [
              Object {
                "padding": undefined,
              },
              undefined,
            ],
            "contentInset": Object {
              "bottom": 0,
            },
            "enableAutomaticScroll": true,
            "enableResetScrollToCoords": true,
            "extraHeight": 75,
            "extraScrollHeight": 0,
            "getScrollResponder": [Function],
            "handleOnScroll": [Function],
            "keyboardDismissMode": "interactive",
            "keyboardOpeningTime": 250,
            "keyboardShouldPersistTaps": "handled",
            "keyboardSpace": 0,
            "onScroll": [Function],
            "resetKeyboardSpace": [Function],
            "resetScrollToCoords": Object {
              "x": 0,
              "y": 0,
            },
            "scrollEventThrottle": 1,
            "scrollForExtraHeightOnAndroid": [Function],
            "scrollToEnd": [Function],
            "scrollToFocusedInput": [Function],
            "scrollToPosition": [Function],
            "showsVerticalScrollIndicator": true,
            "style": Object {
              "backgroundColor": "transparent",
              "flex": 1,
            },
            "viewIsInsideTabBar": false,
            "virtual": undefined,
          },
          Object {},
        ],
        Array [
          Object {
            "children": Array [],
            "data": Object {
              "message": "404 data not found",
            },
            "disableVirtualization": false,
            "getItem": [Function],
            "getItemCount": [Function],
            "horizontal": false,
            "initialNumToRender": 10,
            "invertStickyHeaders": undefined,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onContentSizeChange": [Function],
            "onEndReachedThreshold": 2,
            "onLayout": [Function],
            "onMomentumScrollEnd": [Function],
            "onScroll": [Function],
            "onScrollBeginDrag": [Function],
            "onScrollEndDrag": [Function],
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "stickyHeaderIndices": Array [],
            "updateCellsBatchingPeriod": 50,
            "viewabilityConfigCallbackPairs": Array [],
            "windowSize": 21,
          },
          Object {},
        ],
      ],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ScreenHeader
            title="Lista de Médicos"
          />,
          <FlatList
            data={
              Object {
                "message": "404 data not found",
              }
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Lista de Médicos",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "message": "404 data not found",
            },
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "children": Array [
                <ScreenHeader
                  title="Lista de Médicos"
                />,
                <Styled(Container)
                  virtual={undefined}
                >
                  <Styled(Content)
                    virtual={undefined}
                  >
                    <Styled(Spinner)
                      color="#5f4b8b"
                      virtual={undefined}
                    />
                  </Styled(Content)>
                </Styled(Container)>,
              ],
            },
            Object {},
          ],
          Array [
            Object {
              "automaticallyAdjustContentInsets": false,
              "children": <Styled(Spinner)
                color="#5f4b8b"
                virtual={undefined}
              />,
              "contentContainerStyle": Array [
                Object {
                  "padding": undefined,
                },
                undefined,
              ],
              "contentInset": Object {
                "bottom": 0,
              },
              "enableAutomaticScroll": true,
              "enableResetScrollToCoords": true,
              "extraHeight": 75,
              "extraScrollHeight": 0,
              "getScrollResponder": [Function],
              "handleOnScroll": [Function],
              "keyboardDismissMode": "interactive",
              "keyboardOpeningTime": 250,
              "keyboardShouldPersistTaps": "handled",
              "keyboardSpace": 0,
              "onScroll": [Function],
              "resetKeyboardSpace": [Function],
              "resetScrollToCoords": Object {
                "x": 0,
                "y": 0,
              },
              "scrollEventThrottle": 1,
              "scrollForExtraHeightOnAndroid": [Function],
              "scrollToEnd": [Function],
              "scrollToFocusedInput": [Function],
              "scrollToPosition": [Function],
              "showsVerticalScrollIndicator": true,
              "style": Object {
                "backgroundColor": "transparent",
                "flex": 1,
              },
              "viewIsInsideTabBar": false,
              "virtual": undefined,
            },
            Object {},
          ],
          Array [
            Object {
              "children": Array [],
              "data": Object {
                "message": "404 data not found",
              },
              "disableVirtualization": false,
              "getItem": [Function],
              "getItemCount": [Function],
              "horizontal": false,
              "initialNumToRender": 10,
              "invertStickyHeaders": undefined,
              "keyExtractor": [Function],
              "maxToRenderPerBatch": 10,
              "numColumns": 1,
              "onContentSizeChange": [Function],
              "onEndReachedThreshold": 2,
              "onLayout": [Function],
              "onMomentumScrollEnd": [Function],
              "onScroll": [Function],
              "onScrollBeginDrag": [Function],
              "onScrollEndDrag": [Function],
              "renderItem": [Function],
              "scrollEventThrottle": 50,
              "stickyHeaderIndices": Array [],
              "updateCellsBatchingPeriod": 50,
              "viewabilityConfigCallbackPairs": Array [],
              "windowSize": 21,
            },
            Object {},
          ],
        ],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderBottomColor": "#a7a6ab",
              "borderBottomWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 15,
              "right": 0,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "top": 0,
            },
            Object {
              "backgroundColor": "#5f4b8b",
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "Raleway",
                },
              ]
            }
            virtual={undefined}
          >
             
            Lista de Médicos
             
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "height": 1334,
        }
      }
      virtual={undefined}
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            undefined,
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <ActivityIndicator
            animating={true}
            color="#5f4b8b"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "height": 80,
              }
            }
            virtual={undefined}
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;
